File: CS/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;

namespace InheritedUserControl {
    class Product {
        public int ProductID { get; set; }
        public int CategoryID { get; set; }
        public string ProductName { get; set; }
        public int QuantityPerUnit { get; set; }
        public bool Discontinued { get; set; }
        public decimal UnitPrice { get; set; }
    }

    static class DataHelper {
        public static BindingList<Product> GetData(int count = 10) {
            BindingList<Product> list = new BindingList<Product>();
            for(int i = 0; i < count; i++) {
                list.Add(new Product() {ProductID = i, CategoryID =  i % 3, Discontinued = i % 2 == 0, ProductName = string.Format("Product {0}", i), UnitPrice = i * 20  + i});
            }
            return list;
        }
    }
}

File: VB/Product.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Linq
Imports System.Text

Namespace InheritedUserControl
    Friend Class Product
        Public Property ProductID() As Integer
        Public Property CategoryID() As Integer
        Public Property ProductName() As String
        Public Property QuantityPerUnit() As Integer
        Public Property Discontinued() As Boolean
        Public Property UnitPrice() As Decimal
    End Class

    Friend NotInheritable Class DataHelper

        Private Sub New()
        End Sub

        Public Shared Function GetData(Optional ByVal count As Integer = 10) As BindingList(Of Product)
            Dim list As New BindingList(Of Product)()
            For i As Integer = 0 To count - 1
                list.Add(New Product() With { _
                    .ProductID = i, _
                    .CategoryID = i Mod 3, _
                    .Discontinued = i Mod 2 = 0, _
                    .ProductName = String.Format("Product {0}", i), _
                    .UnitPrice = i * 20 + i _
                })
            Next i
            Return list
        End Function
    End Class
End Namespace
